data/src/main/scala/cats/data/Validated.scala

```diff
@@ -0,0 +1,143 @@
+package cats
+package data
+
+import cats.data.Or.{LeftOr, RightOr}
+import cats.data.Validated.{Valid, Invalid}
+
+sealed abstract class Validated[+E, +A] extends Serializable {
```

Comment:
`List(Valid(...), Invalid(...))` will get inferred as `Validated[X]` instead of `Validated[X] with Product`